File: projects/ecommerce/src/app/common/order.ts
import { CartItem } from "../entities/cart-item";
import { Product } from "../entities/product";
import { Address } from "./address";

export class Order {
    billingAddress: Address;
    shippingAddress: Address;
    cartItems:CartItem[];
    subtotal: number;
    shipping: number;
    total: number;
    modeOfPayment: string;
}

File: projects/angulat tutorials frontend/ecommerceBackend2/src/main/java/com/example/demo/entities/Order.java
package com.example.demo.entities;

import java.util.ArrayList;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
@Entity
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
@Table(name="order_table")
public class Order {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	long id;
	@OneToOne
	Address billingAddress;
	@OneToOne
	Address shippingAddress;
	
	double totalPrice;
	@OneToMany
	ArrayList<Product> products;
}
